portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,21 +35,38 @@
 public class DataServlet extends HttpServlet {
   private List<String> comments;
   private Gson gson;
+  private DatastoreService datastore;
   @Override
   public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
     gson = new Gson();
-    comments = new ArrayList<>();
+    
+    
   }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numComments = getNumParameter(request, "numComments");
+    comments = new ArrayList<>(numComments);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity comment : results.asIterable()) {
+      if (comments.size() == numComments) {
```

Comment:
is the number of comments in results.asIterable() == numComments? In other words, do we need to explicitly add this check?

Reply:
I think I named the numComments variable poorly. It should be the maximum number of comments that the user wants from the server which is why I am breaking the for loop once the comments array has the max number of comments from the query. Will rename numComments to maxNumComments.